tests/components/generic/test_config_flow.py

```diff
@@ -948,40 +890,32 @@ async def test_form_options_permission_error(
 
 
 @pytest.mark.usefixtures("fakeimg_png")
-async def test_unload_entry(hass: HomeAssistant) -> None:
+async def test_unload_entry(hass: HomeAssistant, setup_entry: MockConfigEntry) -> None:
     """Test unloading the generic IP Camera entry."""
-    mock_entry = MockConfigEntry(domain=DOMAIN, options=TESTDATA)
-    mock_entry.add_to_hass(hass)
-
-    await hass.config_entries.async_setup(mock_entry.entry_id)
-    await hass.async_block_till_done()
-    assert mock_entry.state is ConfigEntryState.LOADED
+    assert setup_entry.state is ConfigEntryState.LOADED
```

Comment:
This test should belong in `test_init.py`

Reply:
7cc71e9354602572444080693fa77f8363f36964